packages/flutter_driver/lib/src/driver/driver.dart

```diff
@@ -480,19 +480,11 @@ class FlutterDriver {
 
     // If the item is already visible then we're done.
     bool isVisible = false;
-    try {
-      await waitFor(item, timeout: const Duration(milliseconds: 100));
-      isVisible = true;
-    } on DriverError {
-      // Assume that that waitFor timed out because the item isn't visible.
-    }
-
-    if (!isVisible) {
-      waitFor(item, timeout: timeout).then((Null _) { isVisible = true; });
-      while (!isVisible) {
-        await scroll(scrollable, dxScroll, dyScroll, const Duration(milliseconds: 100));
-        await new Future<Null>.delayed(const Duration(milliseconds: 500));
-      }
+    waitFor(item, timeout: timeout).then((Null _) { isVisible = true; });
```

Comment:
I'll accept all the blame for the implicit rationale for these timeouts and delays. Please add some comments that explain exactly what's going on here, for the sake of the next person who's stuck with trying to understand what's going on...

Reply:
Good idea -- done!